### PR TITLE
NEON implementations of EAC and ETC2 planar compression

### PR DESCRIPTION
**Pull request by hkr**
The alpha compression generates identical results as the reference implementation in C.
The ETC2 planar block mode compression generates slightly different results for some blocks. The floating point calculations are not identical.

I also added the check for solid blocks to ETC2 compression functions.

On a raspberry pi 4 the changes speed up ETC2 RGBA compression up by around 8x.

Code formatting doesn't match the rest of the code, I could clean that up if you are interested.

### CODE REVIEW

File: ProcessRGB.cpp [hunk 1]
```diff
@@ -1902,6 +2117,75 @@ static etcpak_force_inline int GetMulSel( int sel )
...
+#ifdef __ARM_NEON
+
+static constexpr etcpak_force_inline int GetMulSel(int sel)
+{
+    return (sel < 1) ? 0 : (sel < 4) ? 1 : (sel < 5) ? 2 : (sel < 8) ? 3 : (sel < 14) ? 4 : 5;
+}
+
+static constexpr int ClampConstant(int x, int min, int max)
+{
+    return x < min ? min : x > max ? max : x;
```
Comment (wolfpld) [inline ProcessRGB.cpp:2132]: Adapting code from `clampu8()` (Math.hpp:74) might result in faster code.

**Comment by wolfpld** [comment]
Please adapt code formatting where applicable and add yourself to the AUTHORS file.

File: ProcessRGB.cpp (hunk 1 above)
Comment (wolfpld) [reply ProcessRGB.cpp:2132]: (Well, probably not, as it's compile time constant...)

Comment (hkr) [reply ProcessRGB.cpp:2132]: This is just used as a workaround to make it compile after I removed use of `if constexpr` (which requires a compiler that supports C++17).

**Pull request merged**